src/http.cpp

```diff
@@ -1164,6 +1168,12 @@ Request::init_parser()
     };
 }
 
+bool is_listen(const std::string& path)
+{
+    std::string suffix="/listen";
+    return path.size() >= suffix.size() && path.compare(path.size()-suffix.size(), suffix.size(), suffix) == 0;
+}
+
```

Comment:
http.h/cpp is meant to be a generic http layer, this logic should be in the proxy client